### DISCUSSION TITLE
With Graphite Mode Obsolete What Metrics System Should NB5.21 Provide for Standalone Operation

### DISCUSSION THREAD

**Discussion by dave2wave** [nosqlbench/nosqlbench#1979] [General]
While configuring for a default / standalone NB5.21 environment there are options / potential options for the default metrics envrionment. Possibilities might include some of the following options:
- Set up a docker compose of grafana and victoria metrics with standard dashboards and generated local apikeys configured for the local `.nosqlbench` setup.
- Set up a grafana in docker with standard dashboards and using a local mysql or sqllite database.
- Set up a docker compose of nb5.21, grafana, and victoria metrics with standard metrics and self-contained generated apikeys.
- Other ideas?

Please discuss so that effort can be made in the way that is most helpful to the community.

**Comment by jshook** [comment]
I strongly favor the third option, because it keeps us from having to hoist in the docker management layer which we previously found did not play well with other dependencies. It also allows us to delegate out to the system via exec calls and achieve the same result. And further, it provides a recipe for those who simply want to stand up the metrics stack on their own.

**Comment by jshook** [comment]
NB 5.25 exports metrics in a uniform model to local sqlite as a standard rule, while maintaining data to all other concurrent reporters.
